### ISSUE TITLE
Update Inferno driver

### ISSUE THREAD

**Issue by devnoname120** [TheOfficialFloW/Adrenaline#103]
ProCFW's inferno driver originally was a reverse engineering of Sony's NP9660 that fixed a lot of bugs so that it would always run correctly.

It was nonetheless not perfect, and didn't have great performances with some compressed games like GTA Liberty City Stories.

@codestation investigated, and made two changes that overall drastically improved the performances:
- Improve block reading so that only the exact needed blocks are read and decompressed, this provided a huge boost to the performances, making many compressed games playable when they weren't.
- Add LZ4 support (.zso format): this one improved performances a bit, but it was not as noticeable as the other changes.

Here is the patch that introduced these changes: https://github.com/MrColdbird/procfw/commit/4d523e1ef99a7a8e3b29435f07f22c58c819d44b

Please note that these changes are not present in ARK (I think that they were made after ARK's release).

**Comment by TheOfficialFloW** [comment]
Be free to work on this, I don't have the time.

**Comment by devnoname120** [comment]
I'll focus on my projects so I won't be able to do it.

**Comment by devnoname120** [comment]
Apparently Inferno's performances are already good on the PS Vita so there is no need to update it.

**Comment by codestation** [comment]
Just for completeness, i already finished the patch: https://github.com/codestation/Adrenaline/commit/e9c7b0176c4f1e86415ef987aed5431a09871c10

I removed the LZ4 parts since probably nobody is using zso anyway.

Some points about the code:

* The new read method can be disabled by defining `CSO_DEFAULT_READ`.
* The iso cache, cache policy and size are disabled since they aren't present on the AdrenalineConfig struct, see [1] for more info.
* I noticed that some parts of Adrenaline are binary-only, merging this could be the first step to make all of Adrenaline open source.

[1] https://github.com/codestation/Adrenaline/blob/master/cef/inferno/main.c#L110

If you want to merge this but need aditional changes, then let me know.

**Comment by TheOfficialFloW** [comment]
It's fine. can you update the binary file though?

**Comment by kathampy** [comment]
Has this been merged?

**Comment by 133794m3r** [comment]
I don't think this thing's been merged has it? I see the option in the files but the rest of them aren't there.

Also I am one of the only people who used zso but so long as the new inferno driver is included I can live without it. The pull request was closed and didn't seem to be merged.

**Comment by codestation** [comment]
LZ4 was removed because it made no difference on the Vita. The read speeds on games like GOW and GTA are fine with CSO so there is no drawback like on PSP. Also it required additional patches on the xmb for the zso games to show and didn't have time.

If flow want to merge it I recommend to define CSO_DEFAULT_READ , because i tested the patch but for some reason some games (like GOW) had micro stutters on the initial block load while using the new read method, i dunno why.

tl;dr: zero improvements for end users (nor lz4 or new read method). Merge with CSO_DEFAULT_READ if want to remove the inferno blob and build it from source.

**Comment by codestation** [comment]
LZ4 was removed because it made no difference on the Vita. The read speeds on games like GOW and GTA are fine with CSO so there is no drawback like on PSP. Also it required additional patches on the xmb for the zso games to show and didn't have time.

If flow want to merge it I recommend to define CSO_DEFAULT_READ , because i tested the patch but for some reason some games (like GOW) had micro stutters on the initial block load while using the new read method, i dunno why.

tl;dr: zero improvements for end users (nor lz4 or new read method). Merge with CSO_DEFAULT_READ if want to remove the inferno blob and build it from source.

**Comment by 133794m3r** [comment]
is inferno's read speed increases already in adrenaline or do I have to merge your fork with the current head and then compile from source?

I guess I can just deal with no more zsos even though I liked it since it's not providing a real difference apparently anymore. Is there even a cso read cache beyond the reading speedups currently?

**Comment by codestation** [comment]
I noticed no difference in GTA but there was micro stutters in GOW GoS (in the part where the cutscene blended with the gameplay) using the new CSO read method. I have no idea why it happens, read alignment issues? Dunno, don't have time to debug that.

**Comment by 133794m3r** [comment]
So basically the new driver didn't speed up the games at all really and in some of them actually made it worse?

**Comment by TheOfficialFloW** [comment]
Added in https://github.com/TheOfficialFloW/Adrenaline/commit/6b4fd3debba0834e08bea0aef63424440f0bbe59 (I didn't define CSO_DEFAULT_READ though).
